src/core/ext/filters/channel_idle/channel_idle_filter.h

```diff
@@ -54,16 +54,20 @@ class ChannelIdleFilter : public ImplementChannelFilter<ChannelIdleFilter> {
     explicit Call(ChannelIdleFilter* filter) : filter_(filter) {
       filter_->IncreaseCallCount();
     }
-    ~Call() { filter_->DecreaseCallCount(); }
+    ~Call() { MaybeDecrement(); }
 
     static const NoInterceptor OnClientInitialMetadata;
     static const NoInterceptor OnServerInitialMetadata;
     static const NoInterceptor OnServerTrailingMetadata;
     static const NoInterceptor OnClientToServerMessage;
     static const NoInterceptor OnServerToClientMessage;
-    static const NoInterceptor OnFinalize;
+    void OnFinalize(const grpc_call_final_info*) { MaybeDecrement(); }
```

Comment:
Won't `OnFinalize()` always be called before the `Call` object is destroyed?  If so, then it seems like we don't need `MaybeDecrement()`, since we can just directly call `filter_->DecreaseCallCount()` here.